tests/acceptance/features/webUISharingInternalGroups/shareWithGroupsEdgeCases.feature

```diff
@@ -40,6 +40,8 @@ Feature: Sharing files and folders with internal groups
       | user1    |
       | user2    |
       | user3    |
+    # workaround to avoid "empty results" dropdown issue
```

Comment:
is the issue this one https://github.com/owncloud/owncloud-design-system/issues/547 ?
why would that problem happen here?

Reply:
we cannot detect an "empty results" situation because the dropdown does not appear **at all** when no results are found, so there is nothing to wait for.

well, we can still wait for the element until timeout but that will slow down the tests.
but the way how "get all results from the dropdown" is written makes it hard to detect an error vs dropdown just decided to not appear. So I preferred this more reliable approach here that doesn't defeat the purpose of the test.

Reply:
> why would that problem happen here?

in this specific test the problem appeared because we are searching for a user "user1" that we expect not to be there, and that produces an empty results situation with no dropdown

adding a user "user11" will pop up the dropdown with the result "user11", so the test can confirm that there is no "user1" in the dropdown

Reply:
can you mark the Scenario with `@issue-owncloud-design-system-547 ` and if possible write it in a way that it will fail when the issue is fixed https://doc.owncloud.com/server/developer_manual/testing/acceptance-tests.html#writing-scenarios-for-bugs

Reply:
that would require another separate test just to assert that the dropdown "no results" is not there ?

Reply:
see this block for more context: https://github.com/owncloud/phoenix/pull/2450/files#diff-c0efb50a13bceb9cb8549aa05fc0d250R369

basically I added a safeguard there in case the dropdown doesn't open. since this is a generic function, we don't know in advance if we should expect a dropdown or not as we don't know if the calling test is wanting to check for "no results" or not. There's also the option of rewriting such tests and use a completely different approach, which is something I didn't want to do here.

Reply:
yes, just to force someone to delete this hack in the code when the bug is fixed and to make sure test is not :snake: -:oil_drum: when the bug is fixed

Reply:
@individual-it another alternative would be to remove the "workaround" comment and assume that this test was written like that on intention, as the addition doesn't hurt.

once we do fix the "more results" case we can optionally remove that extra user anyway, but it doesn't hurt to keep it

Reply:
I don't think the test is snake oil in any of the states. The test result is still valid regardless of "user11" and/or "no results" dropdown. In some other occurrences I like to have some extra users populated in places instead of limiting the test to just the one in question.